### ISSUE TITLE
🚧Refactor GlyphMetrics to readonly record

### ISSUE THREAD

**Issue by CalvinWilkinson** [KinsonDigital/Velaptor#803]
### Complete The Item Below

- [X] I have updated the title without removing the 🚧 emoji.

### Description

Refactor the `GlyphMetrics` struct to a `readonly record struct`.  This will provide performance benefits as well as make the struct data immutable.  When refactoring the code, use the `with` keyword.

This struct holds data required for creating font atlas textures when loading fonts, which dictates how the text will be rendered with a particular font.  

Also, refactor the properties to `public init` properties.   This will provide the ability to set the initial values of the properties as well as keep the structure immutable.

Make sure that the XML docs for the properties have been updated.
Example:
```diff
/// <summary>
- /// Gets or sets the value.
+ /// Gets the value.
/// </summary>
- public uint MyProp { get; set; };
+ public uint MyProp { get; init; }
```

> **Note**
> Please use the full prop definitions as shown above.  Do not use any primary constructors.
> Further changes are coming down the road related to primary constructors and some research needs to be done first.

> **Warning**
> Be careful with refactoring code related to this `struct`.  The impact could entirely break the font creation process and in turn how text is rendered.  This code is especially sensitive because `unsafe` code is used to bit shift values on this struct based on 32bit vs 64bit processor architectures.  This sensitive code is located in the `FontService` class.

> **Warning**
> This issue will introduce https://github.com/KinsonDigital/Velaptor/labels/%F0%9F%A7%A8breaking%20changes due to the `GlyphMetrics` struct being part of the public API.

**Things To Consider:**
- The `IEquatable<T>` is not required for `record`'s.  This is because the `Equals()` method that the interface provides is built into records.
- Methods like the `Equals()` `override bool Equals()` and `GetHashCode()` are built-into records

Delete the `GlyphMetricsTests` file.  The `struct` is so simple combined with the removal of the `Equals()` methods, that there is nothing to test anymore.

### Acceptance Criteria

- [x] The `GlyphMetrics` structure has been refactored to a `readonly record struct`.
- [x] Related code that uses the `struct` has been refactored 
- [x] The `GlyphMetricsTests` file has been deleted.
- [x] XML docs updated where applicable

### ToDo Items

- [X] Change type labels added to this issue.  Refer to the _**Change Type Labels**_ section below.
- [X] Priority label added to this issue.  Refer to the _**Priority Type Labels**_ section below.
- [x] Issue linked to the correct milestone _(if applicable)_.

### Additional Information:

**_<details closed><summary>Change Type Labels</summary>_**

| Change Type         | Label                |
|---------------------|----------------------|
| Bug Fixes           | `🐛bug` |
| Breaking Changes    | `🧨breaking changes` |
| New Feature         | `✨new feature` |
| CICD Changes        | `♻️cicd` |
| Config Changes      | `⚙️config` |
| Performance Improvements | `🏎️performance` |
| Code Doc Changes    | `🗒️documentation/code` |
| Product Doc Changes | `📝documentation/product` |
</details>

**_<details closed><summary>Priority Type Labels</summary>_**

| Priority Type       | Label             |
|---------------------|-------------------|
| Low Priority        | `low priority`    |
| Medium Priority     | `medium priority` |
| High Priority       | `high priority`   |
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

**Comment by map-b** [comment]
Starting work on this.

**Comment by CalvinWilkinson** [comment]
> Starting work on this.

@map-b I have assigned the issue to you.

Please note, I added a note in the description above stating to make sure to keep the prop definitions in the struct and to not use primary constructors.

Cheers!!

**Comment by map-b** [comment]
Of course! I didn't see it helpful in this case yesterday. I don't know if keeping the definition of the constructor although It isn't used?

**Comment by CalvinWilkinson** [comment]
> Of course! I didn't see it helpful in this case yesterday. I don't know if keeping the definition of the constructor although It isn't used?

Thanks.  I appreciate that!!

As of right now, it looks like I will be avoiding primary constructors when it comes to dependency injection, and then data holding types will probably move to primary constructors.  But I need to think about what problems might arise, performance, `.editorconfig` setup for analyzers, etc. 

I am super excited to move to dotnet 8 though!!  

If you are interested in following (watching) issue #810 as I keep adding to it and doing research, you are welcome to.

If you are interested in knowing the state of the entire organization and all of it's projects, you can view the org project [here](https://github.com/orgs/KinsonDigital/projects/11/views/11)
